gemini/function-calling/intro_diy_react_agent.ipynb

```diff
@@ -0,0 +1,1556 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #31.    def get_action_from_func(func_payload, log, role="MODEL"):
</pre><p>See if this helper function is still needed after switching to <code>resp.candidates[0].function_calls</code>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/generative-ai/pull/721/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Reply:
Still needed mostly because the code needed the payload in JSON to print. All good.